Make items selectable -> editable via config

p3root opened the pull request:
With my patch you can select a item and have a bindable item.

tiberiuzuld commented on src/lib/gridster.component.ts:
```diff
@@ -50,7 +57,25 @@ export class GridsterComponent implements OnInit, OnChanges, OnDestroy, Gridster
+
+
+  @ViewChildren(forwardRef(() => GridsterItemComponent))
+  gridsterItems: QueryList<GridsterItemComponent>;
```
why do you need this?

p3root replied:
was my first approach to get to the components. Dead code. Will be removed!

tiberiuzuld commented on src/lib/gridster.component.ts:
```diff
@@ -76,6 +101,23 @@ export class GridsterComponent implements OnInit, OnChanges, OnDestroy, Gridster
+
+    if (this.options.selectable && this.options.selectable.enabled) {
+      this.selectionService.selectItem.subscribe((item: GridsterItemComponentInterface) => {
+        for (const x of this.grid) {
```
why for loop when you can say:
```javascript
if(this.selectedItem){
	item.item.isSelected = false;	
}
```

p3root replied:
Good point, will change that.

tiberiuzuld commented on src/lib/gridsterSelection.service.ts:
```diff
@@ -0,0 +1,9 @@
+
+@Injectable()
+export class GridsterSelectionService {
+    public selectItem = new EventEmitter<GridsterItemComponentInterface>();
```
don't really like EventEmitters last time I played with them where creating memory leaks.
I prefer callbacks

p3root replied:
Agree with you, depends on the case how you use them. I will add an unsubscribe in the ngOnDestroy function.

tiberiuzuld commented on src/lib/gridsterDraggable.service.ts:
```diff
@@ -243,9 +245,25 @@ export class GridsterDraggable {
       this.mousedown();
       this.touchstart();
     }
+    if (!enableDrag) {
```
why this here ?
you are already in dragMove means you are already making a drag. ...

tiberiuzuld commented on src/lib/gridsterDraggable.service.ts:
```diff
@@ -243,9 +245,25 @@ export class GridsterDraggable {
+    } else {
+      this.selection.selectItem.emit(this.gridsterItem);
+    }
+    this.gridster.updateGrid();
```
why need to update grid on selection?
This method only updates the lines grid from the background.
Yeah name is not the best for this method.

tiberiuzuld commented:
Hi @p3root ,
Have some overall questions:
1. Why this needs to be in the library, this can be implemented outside of the library easily?
2. The library is implemented with callbacks not EventEmiters. 
Last time I used those, where creating memory leaks ...
You can test it with the destroy gridster button top right.. and check profiler.
3. You inserted in gridster.component more code than needed, prefer that code to be separate in the selection service maybe.

Thanks

tiberiuzuld commented on src/lib/gridster.component.ts:
```diff
@@ -76,6 +101,23 @@ export class GridsterComponent implements OnInit, OnChanges, OnDestroy, Gridster
+          x.item.isSelected = false;
+          x.isSelected = false;
+        }
+        item.item.isSelected = true;
```
`item.item` object is the object the user gives to the grid.
Don't think is a good idea to pollute that object with grid properties
Use `$item`

p3root commented:
Hey!

1) I think it can be usefull for other users also. Therefore would it be nice to be in the lib.
2) I have removed the callbacks.
3) I have changed that.

p3root commented:
I have implemented it by my own. Maybe you bring such a feature by yourself sometimes.

tiberiuzuld commented:
Added a ticket https://github.com/tiberiuzuld/angular-gridster2/issues/267
Probably after 5.0 release
Thank you.
